### DISCUSSION TITLE
DuckDB confusion around ordering of `where` and `using sample`

### DISCUSSION THREAD

**Discussion by ajfriend** [duckdb/duckdb#3835] [General]
This isn't a bug report, more of a user experience story. I'm not expecting any specific changes, just posting this here hoping that it might be helpful. :)

Here's some code demonstrating an issue that confused me recently: https://gist.github.com/ajfriend/1b334e4bbf1bd3d13a1d6a103f33f074

Going in, my expectation was that the `using sample 5` filtering would happen **after** the `where` clause filtering. So I was confused by the second example, where I got a random number of rows back.

After playing around with it for a bit, I'm guessing that the `using sample 5` filtering actually happens **before** the `where` clause filtering.

Perhaps this is explained well somewhere in the documentation, but as a Python user who isn't used to the syntax diagrams like in https://duckdb.org/docs/sql/samples, this wasn't obvious to me.

Maybe pointing out the clause ordering more aggressively in the docs would help other folks in the future that might run into this issue? (Or maybe I just missed that part of the docs?)

**Comment by orzom411** [comment]
Good Morning,

A workaround is to nest the predicate, for example:

`select v.co_id, v.name 
  from ( 
     select d.co_id, d.name 
       from read_parquet('../somefolder/some*.parquet') as d 
     where d.co_id = '3293851' limit 100 ) as v 
  using sample 10 rows;`

There are costs to nudging it that way, it might work okay for datasets that aren't too large.

Just a thought, hope it helps.
